types/block.go

```diff
@@ -105,7 +98,9 @@ func (b *Block) ValidateBasic() error {
 	return nil
 }
 
-// fillHeader fills in any remaining header fields that are a function of the block data
+// fillHeader fills in any remaining header fields that are a function of the
+// block data NOTE: we expect celestia-app to populate the block DataHash but we
+// populate it here (in celestia-core) to not break existing tests in this repo.
```

Comment:
In that case, wouldn't it make sense to have a test-only version of this function to keep the units tests happy, but yet have this (production) version with only what Celestia needs, for readability and maintainability?